javatests/build/bazel/tests/integration/BazelBaseTestCaseTest.java

```diff
@@ -14,19 +14,201 @@
+            public void describeTo(final Description description) {
+               description.appendText("successful exit code (0)");
+            }
+        };
+    }
+
+    private SelfDescribing commandDescription(final Command cmd) {
+        return description -> {
+            final String newLine = System.getProperty("line.separator");
+            final List<String> logContents = logsOfInternalTests(cmd.getErrorLines()).collect(Collectors.toList());
+            description
+                    .appendText("std-error:\n")
+                    .appendValueList("", newLine, newLine,cmd.getErrorLines());
+            if (!logContents.isEmpty()) {
+            description
+                    .appendText("Contents of internal test logs:\n")
+                    .appendText("*******************************\n")
+                    .appendValueList(newLine, newLine, newLine,
+                            logContents);
+            }
+        };
+    }
+
+    private List<String> contents(final Path workspacePath) {
+        try {
+            return Files.walk(workspacePath).map(Path::toString).collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Stream<String> logsOfInternalTests(final List<String> errorLines) {
+        return errorLines.stream()
+                .filter(line -> line.contains("(see "))
+                .map(line -> line.split("see ")[1].replace(")", ""))
+                .map(Paths::get)
+                .map(logPath -> {
+                    try {
+                        LinkedList<String> logContents = new LinkedList<>(Files.readAllLines(logPath));
+                        logContents.addFirst("Log contents:");
+                        logContents.addFirst(logPath.toString());
+                        logContents.addFirst("Log path:");
+                        return logContents;
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                })
+                .flatMap(Collection::stream);
+    }
+
+    private void setupPassingTest(final String testName) throws IOException {
+        writePassingTestJavaSource(testName);
+        writeTestBuildFile(testName);
+    }
+
+    private void writeTestBuildFile(final String testName) throws IOException {
+        scratchFile("BUILD", "load(\"//:bazel_integration_test.bzl\", \"bazel_java_integration_test\")\n" +
+                "\n" +
+                "bazel_java_integration_test(\n" +
+                "    name = \"" + testName + "\",\n" +
+                "    test_class = \"" + testName + "\",\n" +
+                "    srcs = [\"" + testName + ".java\"],\n" +
+                ")\n");
+    }
+
+    private void loadIntegrationTestRuleIntoWorkspace() throws IOException {
+        setupRuleSkylarkFiles();
+        setupRuleCode();
+        addExternalRepositoryFor("org_junit", "junit-4.11.jar");
```

Comment:
I would use wildcard and globs to avoid encoding the version name here.

Reply:
I agree. Given copyFromRunfiles only copies specific files and doesn't currently supports directories or regular expressions I'd like to wait until I add directory copying which will make this change much easier.
WDYT?